# 77 Centuries of Virat Kohli

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [2]:
df = pd.read_csv('77 Centuries of Virat Kohli.csv')
df.shape
df

Score Out/Not Out       Against  Batting Order  Inn.  Strike Rate  \
0     116         Out     Australia              6     2          NaN   
1     103         Out   New Zealand              5     2          NaN   
2     103         Out       England              5     2          NaN   
3     107         Out     Australia              5     2          NaN   
4     119         Out  South Africa              4     1          NaN   
..    ...         ...           ...            ...   ...          ...   
72    113         Out     Sri Lanka              3     1       129.88   
73    166     Not Out     Sri Lanka              3     1       150.90   
74    186         Out     Australia              4     4          NaN   
75    121         Out   West Indies              4     1          NaN   
76    122     Not Out      Pakistan              3     1       129.78   

                                   Venue             Column1   H/A  \
0                          Adelaide Oval            Adelaide  Away   
1                 M. Chinnaswamy Stadium           Bangalore  Home   
2   Vidarbha Cricket Association Stadium              Nagpur  Home   
3              M. A. Chidambaram Stadium             Chennai  Home   
4                      Wanderers Stadium        Johannesburg  Away   
..                                   ...                 ...   ...   
72                           ACA Stadium            Guwahati  Home   
73      Greenfield International Stadium  Thiruvananthapuram  Home   
74                 Narendra Modi Stadium           Ahmedabad  Home   
75                     Queen's Park Oval       Port of Spain  Away   
76                  R. Premadasa Stadium             Colombo  Away   

          Date Result Format Man of the Match Captain  Unnamed: 14  
0   24-01-2012   Lost   Test               No      No          NaN  
1   31-08-2012    Won   Test              Yes      No          NaN  
2   13-12-2012  Drawn   Test               No      No          NaN  
3   22-02-2013    Won   Test               No      No          NaN  
4   18-12-2013  Drawn   Test               No      No          NaN  
..         ...    ...    ...              ...     ...          ...  
72  10-01-2023    Won    ODI               No      No          NaN  
73  15-01-2023    Won    ODI              Yes      No          NaN  
74  09-03-2023  Drawn   Test              Yes      No          NaN  
75  20-07-2023  Drawn   Test               No      No          NaN  
76  10-09-2023    Won    ODI              Yes      No          NaN  

[77 rows x 15 columns]

In [3]:
df.columns

Index(['Score', 'Out/Not Out', 'Against', 'Batting Order', 'Inn.',
       'Strike Rate', 'Venue', 'Column1', 'H/A', 'Date', 'Result', 'Format',
       'Man of the Match', 'Captain', 'Unnamed: 14'],
      dtype='object')

In [4]:
df.drop('Unnamed: 14',axis=1,inplace=True)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Score             77 non-null     int64  
 1   Out/Not Out       77 non-null     object 
 2   Against           77 non-null     object 
 3   Batting Order     77 non-null     int64  
 4   Inn.              77 non-null     int64  
 5   Strike Rate       48 non-null     float64
 6   Venue             77 non-null     object 
 7   Column1           77 non-null     object 
 8   H/A               77 non-null     object 
 9   Date              77 non-null     object 
 10  Result            77 non-null     object 
 11  Format            77 non-null     object 
 12  Man of the Match  77 non-null     object 
 13  Captain           77 non-null     object 
dtypes: float64(1), int64(3), object(10)
memory usage: 8.5+ KB


In [6]:
df['Date'] =  pd.to_datetime(df['Date'], format='%d-%m-%Y')

In [7]:
df.isnull().sum()

Score                0
Out/Not Out          0
Against              0
Batting Order        0
Inn.                 0
Strike Rate         29
Venue                0
Column1              0
H/A                  0
Date                 0
Result               0
Format               0
Man of the Match     0
Captain              0
dtype: int64

In [8]:
df[df['Strike Rate'].isnull()]

Score Out/Not Out       Against  Batting Order  Inn.  Strike Rate  \
0     116         Out     Australia              6     2          NaN   
1     103         Out   New Zealand              5     2          NaN   
2     103         Out       England              5     2          NaN   
3     107         Out     Australia              5     2          NaN   
4     119         Out  South Africa              4     1          NaN   
5     105     Not Out   New Zealand              4     4          NaN   
6     115         Out     Australia              4     2          NaN   
7     141         Out     Australia              4     4          NaN   
8     169         Out     Australia              4     2          NaN   
9     147         Out     Australia              4     2          NaN   
10    103         Out     Sri Lanka              4     2          NaN   
11    200         Out   West Indies              4     1          NaN   
12    211         Out   New Zealand              4     1          NaN   
13    167         Out       England              4     1          NaN   
14    235         Out       England              4     2          NaN   
15    204         Out    Bangladesh              4     1          NaN   
16    103     Not Out     Sri Lanka              4     3          NaN   
17    104     Not Out     Sri Lanka              4     3          NaN   
18    213         Out     Sri Lanka              4     2          NaN   
19    243         Out     Sri Lanka              4     1          NaN   
20    153         Out  South Africa              4     2          NaN   
21    149         Out       England              4     2          NaN   
22    103         Out       England              4     3          NaN   
23    139         Out   West Indies              4     1          NaN   
24    123         Out     Australia              4     2          NaN   
25    254     Not Out  South Africa              4     1          NaN   
26    136         Out    Bangladesh              4     2          NaN   
74    186         Out     Australia              4     4          NaN   
75    121         Out   West Indies              4     1          NaN   

                                         Venue         Column1   H/A  \
0                                Adelaide Oval        Adelaide  Away   
1                       M. Chinnaswamy Stadium       Bangalore  Home   
2         Vidarbha Cricket Association Stadium          Nagpur  Home   
3                    M. A. Chidambaram Stadium         Chennai  Home   
4                            Wanderers Stadium    Johannesburg  Away   
5                                Basin Reserve      Wellington  Away   
6                                Adelaide Oval        Adelaide  Away   
7                                Adelaide Oval        Adelaide  Away   
8                     Melbourne Cricket Ground       Melbourne  Away   
9                        Sydney Cricket Ground          Sydney  Away   
10                 Galle International Stadium           Galle  Away   
11                 Sir Vivian Richards Stadium     North Sound  Away   
12                              Holkar Stadium          Indore  Home   
13                    ACA-VDCA Cricket Stadium   Visakhapatnam  Home   
14                            Wankhede Stadium          Mumbai  Home   
15  Rajiv Gandhi International Cricket Stadium       Hyderabad  Home   
16                 Galle International Stadium           Galle  Away   
17                                Eden Gardens         Kolkata  Home   
18        Vidarbha Cricket Association Stadium          Nagpur  Home   
19                     Feroz Shah Kotla Ground           Delhi  Home   
20                             SuperSport Park       Centurion  Away   
21                    Edgbaston Cricket Ground      Birmingham  Away   
22                                Trent Bridge      Nottingham  Away   
23      Saurashtra Cricket Association Stadium          Rajkot  Home   
24           

**These all are Test Centuries here Strike Rate is not that important.**

In [9]:
df['Strike Rate'] = df['Strike Rate'].fillna(0)

In [10]:
df.isnull().sum()

Score               0
Out/Not Out         0
Against             0
Batting Order       0
Inn.                0
Strike Rate         0
Venue               0
Column1             0
H/A                 0
Date                0
Result              0
Format              0
Man of the Match    0
Captain             0
dtype: int64

In [11]:
df.duplicated().sum()

0

**Statastical Description for ODIs and T20Is**

In [12]:
df[df['Format']!='Test'][['Score','Strike Rate']].describe()

Score  Strike Rate
count   48.000000    48.000000
mean   121.479167   115.658125
std     18.266339    24.953527
min    100.000000    84.900000
25%    107.750000    96.767500
50%    116.500000   111.105000
75%    128.250000   123.772500
max    183.000000   200.000000

*Observations:*

* Avg. Score of Virat Kohli in ODIs/T20Is is 121 in those matches where he have scored Century.
* His maximum Score in white ball cricket is 183 Runs.
* He have Scored Centuries with the avg.strike rate of 115.65 in limited over format.

**Statastical Distribution for Test Matches**

In [13]:
df[df['Format']=='Test']['Score'].describe()

count     29.000000
mean     150.758621
std       47.842491
min      103.000000
25%      107.000000
50%      139.000000
75%      186.000000
max      254.000000
Name: Score, dtype: float64

*Observations:*
* Virat's Maximum Score in Test cricket is 254
* Avg.score of Virat Kohli in those matches where he reached century is 150

In [14]:
px.pie(values = df.Format.value_counts().values,names=df.Format.value_counts().index,hole=0.5,title="Virat Kohli's Tons")

*Greatest in Every Format: Virat Kohli had Scored Tons in each format, 47 in ODIs, 29 in TEST and 1 in T20I Cricket*

### Virat's Centuries against other Teams

In [15]:
Tonvs = df.groupby(['Against','Format','Result'])['Score'].count().reset_index().rename(columns ={'Score':'Number_of_Centuries'})
px.bar(data_frame=Tonvs,x='Against',y='Number_of_Centuries',color ='Format',text_auto=True,pattern_shape='Result')

**Observations :**
* Virat Scored maximum 16 Centuries against the Giant Aussies out of which 8 were in ODIs and 8 were in TESTs which is followed by Sri Lanka(total = 15, ODIs = 10, TESTs= 5).

* Pakistan,Zimbabwe and Afganistan are the teams against whome Virat haven't scored any Test Hundred.

* Virat had Scored his only T20I Century against Afganistan

### Virat's Centuries Over the years

In [16]:
df = df.sort_values(by=['Date'])

In [17]:
l = []
for i in range(1,78):
    l.append(i)

In [18]:
df['Int_Century_count'] = l

In [19]:
fig = px.line(data_frame=df,x='Date',y='Int_Century_count',markers=True,title="Virat's International Hundreds")
fig.update_layout(title_x=0.5)
fig.show()

In [20]:
df['Date'].value_counts()

2014-12-09    2
2009-12-24    1
2017-11-24    1
2018-08-01    1
2018-02-16    1
             ..
2014-02-26    1
2014-02-14    1
2014-01-19    1
2013-12-18    1
2023-09-10    1
Name: Date, Length: 76, dtype: int64

### MOM while Scoring Centuries

In [21]:
mom_cen = df.groupby(['Format','Man of the Match'])['Score'].count().reset_index()

In [22]:
fig = px.sunburst(mom_cen,path=['Format','Man of the Match','Score'],values='Score')
fig.update_layout(title='Number of Centuries',title_x=0.5)


**Observtions:**

* **ODIs** : Out of his 47 Centuries in ODI he won MOM 32 times.
* **TESTs** : Out of his 29 Centuries in TEST he won MOM 9 times.
* **T20Is** : He was awarded MOM for his Maiden T20I Ton.

### Win Percent and Virat's Centuries

In [23]:
ton_win = df.groupby(['Result','Format'])['Score'].count().reset_index()

In [24]:
fig = px.sunburst(ton_win,path=['Format','Result'],values='Score')
fig.update_traces(textinfo="label+percent parent")
fig.update_layout(title="Winning and Losing pecentage of India in all three Formats w.r.t Virat's Century",title_x=0.5)
fig.show()

**Observations:**
* India had Won 81% of matches when Virat Scored Century in ODIs
* India had Won 45% and Drawn 31% of matches when Virat Scored Century in TESTs

In [25]:
df_results_against = df.groupby(['Format','Result','Against'])['Score'].count().reset_index()

In [26]:
fig = px.sunburst(df_results_against,path=['Format','Result','Against'],values='Score')
fig.update_traces(textinfo="label+percent parent")

**From the above plot one thing we can clearly observe and that is when ever india lost beside Kohli's Century maximum times opponent was Australia. So though Virat had scored maximum Centuries against Ausssies the Percentage of Winning is not maximum against them**

### Kohli's Centuries Aginst Aussies

In [27]:
fig = px.sunburst(df_results_against[df_results_against['Against']=='Australia'],path=['Format','Result'],values='Score')
fig.update_traces(textinfo="label+percent parent")

**Yes, we were Right beside Kohli's Century India have lost 50% of it's Test matches, 38% of ODIs against Australia.**

### Kohli's Centuries as Captain

In [28]:
ton_cap = df[df['Captain']=='Yes'].groupby(['Captain','Format','Against','Result'])['Score'].count().reset_index()
fig = px.sunburst(ton_cap,path=['Captain','Format','Against','Result'],values='Score')
fig.update_traces(textinfo="label+percent parent")
fig.update_layout(title="Virat's Centuries as a captain and winning Percent of India",title_x=0.5)

**Observation:**

When Virat was leading the team:
* He scored maximum 33% of his Centuries against West Indies in ODIs out of which India won 71% of matches.
* He scored maximum 25% of his Centuries against Sri Lanka in TESTs out of which India had won 40% matches where as 40% were ended in Draw.

### Kohli's Centuries at Home vs Away

In [29]:
H_a = df.groupby(['Against','H/A'])['Score'].count().reset_index().rename(columns={'Score':'Centuries'})
fig = px.bar(data_frame=H_a,x='Against',y='Centuries',color='H/A',text_auto=True)
fig.update_layout(title = "Virat's Tons at Home and Away from Home", title_x=0.5)
fig.show()

In [30]:
H_a = df.groupby(['Captain','Format','H/A'])['Score'].count().reset_index().rename(columns={'Score':'Centuries'})
fig = px.bar(data_frame=H_a,x='Captain',y='Centuries',color='H/A',pattern_shape='Format',text_auto=True)
fig.update_layout(title = "Virat's Tons at Home and Away from Home as a Captain", title_x=0.5)
fig.show()